pkg/asset/installconfig/ibmcloud/subnet.go

```diff
@@ -0,0 +1,63 @@
+package ibmcloud
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+)
+
+// Subnet represents an IBM Cloud VPC Subnet
+type Subnet struct {
+	CIDR string
+	CRN  string
+	ID   string
+	Name string
+	VPC  string
+	Zone string
+}
+
+func getSubnets(ctx context.Context, client *Client, region string, subnetNames []string) (map[string]Subnet, error) {
+	subnets := map[string]Subnet{}
+
+	for _, name := range subnetNames {
+		results, err := client.GetSubnetByName(ctx, name, region)
+		if err != nil {
+			return nil, errors.Wrapf(err, "getting subnet %s", name)
+		}
+
+		if results.ID == nil {
+			continue
```

Comment:
what is the impact of results.ID being nil?  does this imply an error or does this imply that a given subnet  name wasnt found?

Reply:
Yeah, it's probably better raising an error like the other checks. I'll update.

Reply:
updated